002_simulation_structure/002_simulation_structure.tex

```diff
@@ -17,8 +17,205 @@ \section{Simulation Flow}\label{simulation_flow}
+The minimum default message had to have the following fields that are accessible to all the agents that receive it:
+\begin{itemize}
+    \item \textbf{\texttt{MessageType}}: Indicates one of 13 enumerated message types.
+    \item \textbf{\texttt{SenderFloor}}: Returns the floor this message was sent from.
+    \item \textbf{\texttt{TargetFloor}}: The floor the message is addressed to.
+    \item \textbf{\texttt{ID}}: Each message sent is given a unique ID.
+\end{itemize}
+\vspace{\baselineskip}
+For the MVP, Team 1 constructed a system where for example in a scenario where Alice (currently at floor 10) wishes to send a message upward:
+\begin{enumerate}
+    \item At Tick 0, Alice would construct the message for Bob containing the following information:
+        \begin{itemize}
+            \item \textbf{Alice's ID}
+            \item \textbf{Alice's current floor}: 30
+            \item \textbf{Message's target floor}: 31 \textit{Since Alice wants to message the floor above her}
+            \item \textbf{Message Type} (to be elaborated on the Common Languages Section)
+        \end{itemize}
+    \item Still at Tick 0, Alice calls \texttt{SendMessage} which takes her message and passes it to the tower.
+    \item The tower acts as the communication authority, finding the agent who is on floor 31 at Tick 0 and inserts the message into the recipient agent's (Bob's) inbox.
+    \item At Tick 1, Bob may choose to call \texttt{ReceiveMessage} which would extract Alice's message (if Alice was the first or only agent to send him a message at Tick 0) and respond.
+    \item If Bob has called \texttt{ReceiveMessage}, Alice's message is removed from Bob's inbox.
+\end{enumerate}
+With this implementation, each agent is responsible for calling \texttt{ReceiveMessage} once per tick to receive any messages. The inbox is FIFO - If multiple messages were received or if there were outstanding messages from previous ticks, only the earliest one can be retrieved at a given tick.
+
+Additionally, ``receiving" does not necessarily imply ``reacting", an agent is capable of receiving a message and not responding to it at all without the sender's knowledge. Decoupling ``receiving" and ``reacting" gave agent teams the freedom to ignore messages entirely (metaphorically covering their ears in the tower and refusing to listen to anybody), to listen to messages and do nothing about them, or to actively listen and react. \newline
+Beyond the MVP, however, messages needed to have meaning. Because the process would be automated it was meaningless to send unstructured strings, how was an agent to understand ``Please only eat what you need to survive."? \newline
+A common language was necessary.
```

Comment:
```suggestion
Beyond the MVP, however, messages needed to have meaning. Because the process would be automated it was meaningless to send unstructured strings. An agent would not be able to understand `Please only eat what you need to survive.". It was decided that a common language was necessary. \newline

```